### ISSUE TITLE
torchaudio.load returns 0 frames from ogg files when provided with random offsets

### ISSUE THREAD

**Issue by draguve** [pytorch/audio#3012]
### 🐛 Describe the bug

torchaudio.load loads 0 frames from ogg files at specific offsets, and on others, it runs fine.
It also isn't the only file, it happens on quite a few different files from my dataset at random offsets that fail to load any frames.

```python
import torchaudio
file_location = "./Downloads/1979_p/1979.ogg"
offset = 6306300

info = torchaudio.info(file_location)
file_sample_rate = info.sample_rate
waveform, sample_rate = torchaudio.load(file_location, normalize=True, frame_offset=offset,
                                                num_frames=file_sample_rate)
print(info)
print(f"sample_rate {sample_rate} shape : {waveform.shape}")

offset += file_sample_rate

waveform, sample_rate = torchaudio.load(file_location, normalize=True, frame_offset=offset,
                                                num_frames=file_sample_rate)
print(f"sample_rate {sample_rate} shape : {waveform.shape}")
```

Which gives the output 

```
AudioMetaData(sample_rate=44100, num_frames=12060992, num_channels=2, bits_per_sample=0, encoding=VORBIS)
sample_rate 44100 shape : torch.Size([2, 0])
sample_rate 44100 shape : torch.Size([2, 44100])
```

### Versions

PyTorch version: 1.13.1
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Home Single Language
GCC version: Could not collect
Clang version: 14.0.6
CMake version: version 3.23.2
Libc version: N/A

Python version: 3.10.9 | packaged by conda-forge | (main, Jan 11 2023, 15:15:40) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22621-SP0
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3070 Laptop GPU
Nvidia driver version: 528.02
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\cudnn_ops_train64_8.dll
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==1.13.1
[pip3] torchaudio==0.13.1
[pip3] torchshape==0.0.8
[pip3] torchvision==0.14.1
[conda] blas                      1.0                         mkl
[conda] mkl                       2021.4.0           haa95532_640
[conda] mkl-service               2.4.0           py310h2bbff1b_0
[conda] mkl_fft                   1.3.1           py310ha0764ea_0
[conda] mkl_random                1.2.2           py310h4ed8f06_0
[conda] numpy                     1.23.5          py310h60c9a35_0
[conda] numpy-base                1.23.5          py310h04254f7_0
[conda] pytorch                   1.13.1          py3.10_cuda11.7_cudnn8_0    pytorch
[conda] pytorch-cuda              11.7                 h67b0de4_1    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.13.1                   pypi_0    pypi
[conda] torchshape                0.0.8                    pypi_0    pypi
[conda] torchvision               0.14.1                   pypi_0    pypi

**Comment by turian** [comment]
@draguve  can you share the ogg?

**Comment by mthrok** [comment]
Hi

I looked into this issue, and it seems that on main branch it is no longer happening.
We changed the way torchaudio integrates libsox in #3497, and now we ask users to install libsox seaprately.

SoX v14.4.2 from apt on Ubuntu worked fine, and I bet this was an issue with vorbis encoder or how libsox integrates vorbis.

Therefore I will close the issue. Feel free to open reopen if it's not the case.
